client/coral-embed-stream/src/containers/Stream.js

```diff
@@ -112,8 +112,8 @@ class StreamContainer extends React.Component {
 
   componentDidMount() {
     if (this.props.previousTab) {
-      this.props.data.refetch();
-      this.props.setCommentCountCache(0);
+      this.props.data.refetch()
+        .then(() => this.props.setCommentCountCache(0));
```

Comment:
The commentCountCache basically should save the `asset.commentCount` from the last full query, which is then used to determine if new comments have arrive when the our count polling returns a higher commentCount.